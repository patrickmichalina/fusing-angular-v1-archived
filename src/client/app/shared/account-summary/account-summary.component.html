<img [src]="actImageUrl" class="avatar">
<div class="mat-typography">
  <h3>{{ actName }}</h3>
  <h4>{{ actEmail }}</h4>
</div>