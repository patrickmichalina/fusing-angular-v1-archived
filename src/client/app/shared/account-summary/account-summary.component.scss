@import '../../../../../node_modules/@angular/material/_theming.scss';
:host {
  padding: 1em;
  display: flex;
  align-items: center;
  border-bottom: solid 1px mat-color($mat-grey, 200);
  div {
    flex-direction: column;
  }
  h3,
  h4 {
    margin: 0;
  }
  h3 {
    font-size: 16px;
    text-align: center;
  }
  h4 {
    font-weight: 100;
    font-size: 12px;
  }
  img {
    margin-right: 0.8em;
  }
}
