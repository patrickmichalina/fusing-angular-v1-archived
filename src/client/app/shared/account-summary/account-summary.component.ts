import { ChangeDetectionStrategy, Component, Input } from '@angular/core'

@Component({
  selector: 'pm-account-summary',
  templateUrl: './account-summary.component.html',
  styleUrls: ['./account-summary.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AccountSummaryComponent {
  @Input() readonly actImageUrl: string | undefined
  @Input() readonly actEmail: string | undefined
  @Input() readonly actName: string | undefined
}
