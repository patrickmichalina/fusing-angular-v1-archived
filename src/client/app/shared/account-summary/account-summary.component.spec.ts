import { AccountSummaryComponent } from './account-summary.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'

describe(AccountSummaryComponent.name, () => {
  let fixture: ComponentFixture<AccountSummaryComponent>

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        declarations: [AccountSummaryComponent, TestComponent]
      }).compileComponents()
    })
  )

  beforeEach(() => {
    fixture = TestBed.createComponent(AccountSummaryComponent)
  })

  afterEach(() => {
    TestBed.resetTestingModule()
  })

  it(
    'should compile',
    async(() => {
      fixture.detectChanges()
      expect(fixture.nativeElement).toBeDefined()
      expect(fixture.nativeElement).toMatchSnapshot()
    })
  )
})

@Component({
  selector: 'test-component',
  template: '<pm-account-summary></pm-account-summary>'
})
class TestComponent {}
