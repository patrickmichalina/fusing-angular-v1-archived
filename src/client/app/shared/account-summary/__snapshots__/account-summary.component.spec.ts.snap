// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSummaryComponent should compile 1`] = `
<div
  id="root0"
  ng-version="6.0.0-rc.2"
>
  <img
    class="avatar"
    src="null"
  />
  <div
    class="mat-typography"
  >
    <h3>
      
    </h3>
    <h4>
      
    </h4>
  </div>
</div>
`;
